### PR TITLE
CodeQL plugin filter support [Rebase & FF]

### PR DESCRIPTION
**Pull request by makubacki**
## Description

**Updates readme as of this change**

```
Plugin/CodeQL: Add plugin advantages to readme

Updates the beginning of the readme file to describe the advantages
of the plugin for new readers.
```

**Adds filtering support to the CodeQL plugin**

```
Adds the ability to filter issues with filter patterns defined in
an arbitrary number of YAML files and/or at individual package level
with a section in the package CI YAML file.

Further details are available in the Readme.md file.
```

- [x] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [x] Includes documentation?

## How This Was Tested

- Verified functionality with no filtering (no regression)
- Verified filter all rules in a file
- Verified filter specific rules in a file
- Verified filter against multiple files
- Verified filter file
- Verified multiple filter files
- Verified multiple filter files + filters in package CI YAML file
- Verified malformed filter files report errors

## Integration Instructions

View the instructions in `.pytool/Plugin/CodeQL/Readme.md` in this change.

Signed-off-by: Michael Kubacki <michael.kubacki@microsoft.com>

### CODE REVIEW

File: .pytool/Plugin/CodeQL/Readme.md [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
 
+While CodeQL can be run in a CI environment with other approaches. This plugin offers the following advantages:
+
+1. Provides exactly the same results locally as on a CI server.
+2. Integrates very well into VS Code.
+3. Very simple to use - just use normal Stuart update and build commands.
+4. Very simple to understand - minimally wraps the official CodeQL CLI.
+5. Very simple to integrate - works like any other Stuart build plugin.
+   - Integration is usually just a few lines of code.
+6. Portable - not tied to Azure DevOps specific, GitHub specific, or other host infrastructure.
```
Comment (apop5) [inline .pytool/Plugin/CodeQL/Readme.md:16]: 7. Reproduceable results too. the versioning of queries being integrated into mu help create reproducible results for a project.

File: .pytool/Plugin/CodeQL/analyze/globber.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
+# @file globber.py
+#
+# Provides global functionality for use by the CodeQL plugin.
```
Comment (apop5) [inline .pytool/Plugin/CodeQL/analyze/globber.py:3]: given the apache license, are there any license files that need to be distributed with this?

**Review by apop5 — APPROVED** [review]

File: .pytool/Plugin/CodeQL/Readme.md (hunk 1 above)
Comment (makubacki) [reply .pytool/Plugin/CodeQL/Readme.md:16]: That's a good point. I added 7 to mention reproducibility.

File: .pytool/Plugin/CodeQL/analyze/globber.py (hunk 2 above)
Comment (makubacki) [reply .pytool/Plugin/CodeQL/analyze/globber.py:3]: I don't think so. I've linked to the Apache license file in the source repo and left copyright headers intact. I did not see a NOTICES file to bring over and the license permits relicensing in derivative works, so I think it's okay.

**Review by kuqin12 — APPROVED** [review]

**Pull request merged**